src/script/ptl-tool.py

```diff
@@ -1,27 +1,84 @@
+#!/usr/bin/env python2
+
+# README:
+#
+# This tool's purpose is to make it easier to merge PRs into Ceph.
+#
+# Because developers often have custom names for the ceph upstream remote
+# (https://github.com/ceph/ceph.git), You will probably want to set the
+# PTL_TOOL_BASE_PATH environment variable in your shell rc files before using
+# this script:
+#
+#     PTL_TOOL_BASE_PATH=refs/remotes/upstream/
+#
+#
+# ** Here are some basic exmples to get started: **
+#
+# Merging PR #1234567 and #2345678 into a new test branch with a testing label added to the PR:
+#
+# $ env PTL_TOOL_BASE_PATH=refs/remotes/upstream/ src/scripts/ptl-tool.py --base master 1234567 2345678
+# Detaching HEAD onto base: master
+# Merging PR #1234567
+# Labeled PR #1234567 wip-pdonnell-testing
+# Merging PR #2345678
+# Labeled PR #2345678 wip-pdonnell-testing
+# Deleted old test branch wip-pdonnell-testing-20170928
+# Created branch wip-pdonnell-testing-20170928
+# Created tag testing/wip-pdonnell-testing-20170928_03
+#
+#
+# Merging PR #1234567 into master leaving a detached HEAD (i.e. do not update your repo's master branch) and do not label:
+#
+# $ env PTL_TOOL_BASE_PATH=refs/remotes/upstream/ src/scripts/ptl-tool.py --base master --branch HEAD --merge-branch-name --label - master 1234567
+# Detaching HEAD onto base: master
+# Merging PR #1234567
+# Leaving HEAD detached; no branch anchors your commits
+#
+# Now push to master:
+# $ git push upstream HEAD:master
+#
+#
+# Merging PR #12345678 into luminous leaving a detached HEAD (i.e. do not update your repo's master branch) and do not label:
+# $ env PTL_TOOL_BASE_PATH=refs/remotes/upstream/ src/scripts/ptl-tool.py --base luminous --branch HEAD --merge-branch-name luminous --label - 12345678
+# Detaching HEAD onto base: luminous
+# Merging PR #12345678
+# Leaving HEAD detached; no branch anchors your commits
+#
+# Now push to luminous:
+# $ git push upstream HEAD:luminous
+
+
 # TODO
+# Fetch PRs by label.
 # Look for check failures?
 # redmine issue update: http://www.redmine.org/projects/redmine/wiki/Rest_Issues
 
+import argparse
 import datetime
 import getpass
 import git
```

Comment:
$ pip install gitpython -> add it in install-deps.sh ?

Reply:
No, this is something you should install manually. We don't want that installed for users of Ceph.

Reply:
Oh yes, Sorry my bad :-)